shared/ui-library/src/lib/styled-context/styled-context.tsx

```diff
@@ -0,0 +1,78 @@
+import { createContext, type FC, type ReactNode, useCallback, useEffect, useState } from "react";
+import { ThemeProvider } from "styled-components";
+
+enum ThemeOptions {
+  Light = "light",
+  Dark = "dark"
+}
+
+interface StyledContextProps {
+  activeTheme: ThemeOptions;
+  updateActiveTheme: (newTheme: ThemeOptions) => void;
+}
+
+const StyledContext = createContext<StyledContextProps>({
+  activeTheme: ThemeOptions.Light,
+  updateActiveTheme: () => { /* do nothing */
+  }
+});
+
+const themes = {
```

Comment:
Move to a data / constant file. This could get fairly complex, if the theme is developed further.